### DISCUSSION TITLE
exclude certain endpoints in resetApiState

### DISCUSSION THREAD

**Discussion by bryanprimus** [reduxjs/redux-toolkit#3285] [Q&A]
Is there a way to exclude certain endpoints from `resetApiState` util?

my app has a simulation feature. when the user starts the simulation. backend will start sending simulation data. 

the problem is the network request that was already made through rtk query will be cached. I need to reset all the cache data and then I found `resetApiState`, but I want to exclude some endpoints (in this case authEndpoints) so that it won't get reset.

for now, I ended up separating `authApi` from `mainApi` and only calling resetApiState with `mainApi`

```ts
import { createApi } from '@reduxjs/toolkit/query/react'

const authApi = createApi({})

const mainApi = createApi({})

store.dispatch(mainApi.util.resetApiState());
```

but I believe as mentioned here it is not recommended way. we should only have one API slice for the rtk query

https://redux.js.org/tutorials/essentials/part-7-rtk-query-basics#defining-an-api-slice
https://github.com/reduxjs/redux-toolkit/issues/1725#issuecomment-1451329094

**Comment by markerikson** [comment]
No, `resetApiState` is designed to remove everything.  There's been a number of requests for the ability to remove a single cache entry, but we haven't worked on that specifically.

(There _is_ an internal "remove" action, but I'm not sure if it actually does all the cleanup you might need in such a case.)

**Reply by noveogroup-amorgunov** [reply]
@markerikson hello. It may be useful, when we use `injectEndpoints` method for splitting api into modules. For example, there are two modules (in real project we have about 50 modules):

```ts
// modules/cart/api.ts
export const cartApi = baseApi.injectEndpoints({ ... })

// modules/user/api.ts
export const userApi = baseApi.injectEndpoints({ ... })

// ...
```

and I want to clear only `cartApi` cache with `cartApi.util.resetApiState()`, but change will be reset in all `baseApi`. Do you have any ideas (or recommendation), how reset cache from endpoints only in `cartApi` ?

**Reply by EskiMojo14** [reply]
out of curiosity, is tag invalidation not what you want? It'll remove any queries that don't have any subscriptions left, and will refetch the ones that do.
